src/backend/distributed/planner/query_pushdown_planning.c

```diff
@@ -1148,11 +1165,28 @@ DeferErrorIfUnsupportedTableCombination(Query *queryTree)
 		 */
 		if (rangeTableEntry->rtekind == RTE_RELATION ||
 			rangeTableEntry->rtekind == RTE_SUBQUERY ||
-			rangeTableEntry->rtekind == RTE_RESULT
-			)
+			rangeTableEntry->rtekind == RTE_RESULT)
 		{
 			/* accepted */
 		}
+		else if (rangeTableEntry->rtekind == RTE_VALUES)
+		{
+			int valuesRowCount = list_length(rangeTableEntry->values_lists);
+			if (ValuesMaterializationThreshold > 0 &&
```

Comment:
should this be >= 0?

0 seems like it should mean "materialize everything"

Reply:
Yes, that's a good point.